### PR TITLE
soc/intel_adsp: add support for Intel Jasper Lake

### PR DESCRIPTION
**Pull request by kv2019i**
Add a variant of Intel cAVS2.0 used in Jasper Lake based
products.

BugLink: https://github.com/thesofproject/sof/issues/4539

Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
Just curious, can you summarize the differences with cAVS 2.0? The duplication makes it hard to see.

File: soc/xtensa/intel_adsp/common/boot.c [hunk 1]
```diff
@@ -45,7 +45,11 @@
 #elif defined(CONFIG_SOC_SERIES_INTEL_CAVS_V20)
 #define PLATFORM_INIT_HPSRAM
 #define PLATFORM_INIT_LPSRAM
+#if defined(CONFIG_BOARD_INTEL_ADSP_CAVS20_JSL)
+#define PLATFORM_HPSRAM_EBB_COUNT 16
```
Comment (andyross) [inline soc/xtensa/intel_adsp/common/boot.c:49]: This is another situation where we've got some legacy hardware parameters still sitting in C code.  Ideally we want this value to be an attribute on a devicetree node associated with hpsram (which we have, but all it stores right now is the byte size).

**Review by andyross — APPROVED** [review]

File: soc/xtensa/intel_adsp/common/boot.c (hunk 1 above)
Comment (kv2019i) [reply soc/xtensa/intel_adsp/common/boot.c:49]: @andyross I was thinking the same. I can take a look at replacing this with a DT look-up (in a separate patch).

**Comment by kv2019i** [comment]
> Just curious, can you summarize the differences with cAVS 2.0? The duplication makes it hard to see.

2 cores instead of 4
16 HP-SRAM banks instead of 47
no ALH interface

**Review by nashif — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
